receiver/snmpreceiver/scraper.go

```diff
@@ -39,13 +64,345 @@ func newScraper(logger *zap.Logger, cfg *Config, settings component.ReceiverCrea
+func (s *snmpScraper) scalarDataToMetric(
+	data SNMPData,
+	metricHelper *otelMetricHelper,
+	configHelper *configHelper,
+) error {
+	// Get the related metric name for this SNMP indexed data
+	metricName := configHelper.getMetricName(data.oid)
+
+	// Keys will be determined from the related attribute config and enum values will come straight from
+	// the metric config's attribute values.
+	dataPointAttributes := getScalarDataPointAttributes(configHelper, data.oid)
+
+	return addMetricDataPointToResource(data, metricHelper, configHelper, metricName, generalResourceKey, dataPointAttributes)
+}
+
+// indexedDataToMetric will take one piece of column OID SNMP indexed metric data and turn it
+// into a datapoint for either a new or existing metric with attributes that belongs to either
+// a new or existing resource
+func (s *snmpScraper) indexedDataToMetric(
+	data SNMPData,
+	metricHelper *otelMetricHelper,
+	configHelper *configHelper,
+	indexedAttributeValues map[indexedAttributeKey]string,
+	indexedResourceAttributeValues map[indexedAttributeKey]string,
+) error {
+	// Get the related metric name for this SNMP indexed data
+	metricName := configHelper.getMetricName(data.parentOID)
+
+	indexString := strings.TrimPrefix(data.oid, data.parentOID)
+
+	// Get data point attributes
+	dataPointAttributes, err := getIndexedDataPointAttributes(configHelper, data.parentOID, indexString, indexedAttributeValues)
+	if err != nil {
+		return fmt.Errorf(errMsgOIDAttributeEmptyValue, metricName, err)
+	}
+
+	// Get resource attributes
+	resourceAttributes, err := getResourceAttributes(configHelper, data.parentOID, indexString, indexedResourceAttributeValues)
+	if err != nil {
+		return fmt.Errorf(errMsgOIDResourceAttributeEmptyValue, metricName, err)
+	}
+
+	// Create a resource key using all of the relevant resource attribute names along
+	// with the row index of the SNMP data
+	resourceAttributeNames := configHelper.getResourceAttributeNames(data.parentOID)
+	resourceKey := getResourceKey(resourceAttributeNames, indexString)
+
+	// Create a new resource if needed
+	resource := metricHelper.getResource(resourceKey)
+	if resource == nil {
+		metricHelper.createResource(resourceKey, resourceAttributes)
+	}
 
-	return md, nil
+	return addMetricDataPointToResource(data, metricHelper, configHelper, metricName, resourceKey, dataPointAttributes)
+}
+
+func addMetricDataPointToResource(
+	data SNMPData,
+	metricHelper *otelMetricHelper,
+	configHelper *configHelper,
+	metricName string,
+	resourceKey string,
+	dataPointAttributes map[string]string,
+) error {
+	// Return an error if this SNMP indexed data is not of a useable type
+	if data.valueType == notSupportedVal || data.valueType == stringVal {
+		return fmt.Errorf(errMsgBadValueType, data.oid)
+	}
+
+	// Get the related metric config
+	metricCfg := configHelper.getMetricConfig(metricName)
+
+	// Create a new metric if needed
+	metric := metricHelper.getMetric(resourceKey, metricName)
+	if metric == nil {
```

Comment:
Since we don't intend to use it other than to validate its presence.
```suggestion
	if metric := metricHelper.getMetric(resourceKey, metricName); metric == nil {
```